File: src/main/cs/Problem0704.cs
using System;
using System.Diagnostics;

public class Problem0704
{
    public Problem0704()
    {
        Stopwatch stopWatch = new Stopwatch();
        stopWatch.Start();
        UInt64 result = 0;

        // Solution:
        UInt64 max = 10000000;
        int lastPercent = 0;
        for (UInt64 n = 1; n <= max; n++)
        {
            double percent = (double) n / max * 100;
            if (percent * 100 > lastPercent)
            {
                System.Console.Write("\r" + (percent).ToString("0.##") + "%");
                lastPercent = (int) (percent * 100);
            }

            result += f(n);
        }

        stopWatch.Stop();
        string test = stopWatch.Elapsed.ToString();
        Console.WriteLine("Result:\t" + result + "\tTime:\t" +
                          (double.Parse(test.Substring(test.LastIndexOf(":") + 1, 2)) >= 1
                              ? double.Parse(test.Substring(test.LastIndexOf(":") + 1)) + " s"
                              : double.Parse(test.Substring(test.IndexOf(".") + 1)) / 10_000 + " ms"));
    }

    private UInt64 f(UInt64 n)
    {
        UInt64 result = 0;
        for (UInt64 m = 0; m <= n; m++)
        {
            UInt64 g = this.g(n, m);
            if (g > result)
            {
                result = g;
            }
        }

        return result;
    }

    private UInt64 g(UInt64 n, UInt64 m)
    {
        // n over m
        n = fac(n, n - m) / fac(m);

        // find biggest potenz of 2
        UInt64 max = 0;
        UInt64 j = 1;
        for (UInt64 i = 0; j < n; i++, j = j * 2)
        {
            if (n % j != 0)
            {
                return max;
            }

            max = i;
        }

        return max;
    }

    private UInt64 fac(UInt64 n)
    {
        if (n == 0) return 1;
        for (UInt64 i = n - 1; i > 1; i--)
        {
            n = n * i;
        }

        return n > 0 ? n : 1;
    }

    private UInt64 fac(UInt64 n, UInt64 m)
    {
        if (n == 0) return 1;
        for (UInt64 i = n - 1; i > m; i--)
        {
            n = n * i;
        }

        return n > 0 ? n : 1;
    }
}

File: src/main/java/Problem0704.java
import java.math.BigInteger;

public class Problem0704 {
    public Problem0704() {
        long startTime = System.nanoTime();
        long result = 0;

        // Solution:
        for (long n = 1; n <= 10000000L; n++) {
            result += this.f(n);
        }

        long timeToResolve = System.nanoTime() - startTime;
        System.out.println("Result:\t" + result + "\tTime:\t" + (((double) timeToResolve / 1_000_000) > 1000 ?
                (((double) timeToResolve / 1_000_000_000) + "s") :
                (((double) timeToResolve / 1_000_000) + "ms")));
    }

    private long f(long n) {
        long result = 0;
        for (int m = 0; m <= n; m++) {
            long g = this.g(n, m);
            if (g > result) {
                result = g;
            }
        }
        return result;
    }

    private long g(long lN, long m) {
        // n over m
        BigInteger n = this.fac(lN, lN - m).divide(this.fac(m));

        // find biggest potenz of 2
        long max = 0;
        BigInteger j = BigInteger.ONE;
        for (long i = 0; j.compareTo(n) < 0; i++, j = j.multiply(BigInteger.TWO)) {
            if (n.mod(j).compareTo(BigInteger.ZERO) != 0) {
                return max;
            }
            max = i;
        }
        return max;
    }

    private BigInteger fac(long n) {
        if (n == 0) return BigInteger.ONE;
        BigInteger bN = new BigInteger(n + "");
        for (long i = n - 1; i > 1; i--) {
            bN = bN.multiply(new BigInteger(i + ""));
        }
        return bN.compareTo(BigInteger.ZERO) > 0 ? bN : BigInteger.ONE;
    }

    private BigInteger fac(long n, long m) {
        if (n == 0) return BigInteger.ONE;
        BigInteger bN = new BigInteger(n + "");
        for (long i = n - 1; i > m; i--) {
            bN = bN.multiply(new BigInteger(i + ""));
        }
        return bN.compareTo(BigInteger.ZERO) > 0 ? bN : BigInteger.ONE;
    }
}
